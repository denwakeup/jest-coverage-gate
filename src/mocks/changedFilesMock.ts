import { EDiffFileStatus } from '../constants/coverageGate.constants';

export const rootDirMock = '/Users/packages/coverage-gate/src';

export const changedFilesMock = new Map(
  Object.entries({
    [`${rootDirMock}/apps/simple/src/utils/parseData.ts`]: {
      filePath: `${rootDirMock}/apps/simple/src/utils/parseData.ts`,
      status: EDiffFileStatus.MODIFIED,
    },
    [`${rootDirMock}/apps/simple/src/components/Header/index.tsx`]: {
      filePath: `${rootDirMock}/apps/simple/src/components/Header/index.tsx`,
      status: EDiffFileStatus.ADDED,
    },
  })
);
