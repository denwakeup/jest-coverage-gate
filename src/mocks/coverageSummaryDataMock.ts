import type { CoverageSummaryData } from 'istanbul-lib-coverage';

export const coverageSummaryDataMock: Partial<CoverageSummaryData> = {
  lines: {
    total: 30,
    covered: 20,
    skipped: 10,
    pct: 60,
  },
  branches: {
    total: 30,
    covered: 20,
    skipped: 10,
    pct: 60,
  },
  functions: {
    total: 30,
    covered: 20,
    skipped: 10,
    pct: 60,
  },
};
