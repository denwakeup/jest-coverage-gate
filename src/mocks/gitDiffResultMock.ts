export const gitDiffResultMock = `
M\tapps/simple/src/utils/parseData.ts
A\tapps/simple/src/components/Header/index.tsx
incorrectValue
`;
