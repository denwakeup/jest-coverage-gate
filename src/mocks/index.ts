export * from './changedFilesMock';
export * from './gitDiffResultMock';
export * from './diffCoverageCheckResultMock';
export * from './coverageSummaryDataMock';
