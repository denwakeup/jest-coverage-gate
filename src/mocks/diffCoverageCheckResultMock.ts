import {
  ECoverageCheckStatus,
  EDiffFileStatus,
} from '../constants/coverageGate.constants';
import { IDiffCoverageCheckResult } from '../interfaces/coverage.interfaces';

import { rootDirMock } from './changedFilesMock';

export const diffCoverageCheckResultMock: IDiffCoverageCheckResult = {
  status: ECoverageCheckStatus.ERROR,
  summary: new Map(
    Object.entries({
      [`${rootDirMock}/apps/simple/src/utils/parseData.ts`]: {
        file: {
          filePath: `${rootDirMock}/apps/simple/src/utils/parseData.ts`,
          status: EDiffFileStatus.MODIFIED,
        },
        status: ECoverageCheckStatus.SUCCESS,
        summary: {
          functions: {
            expected: 50,
            current: 80,
            status: ECoverageCheckStatus.SUCCESS,
          },
        },
      },
      [`${rootDirMock}/apps/simple/src/components/Header/index.tsx`]: {
        file: {
          filePath: `${rootDirMock}/apps/simple/src/components/Header/index.tsx`,
          status: EDiffFileStatus.ADDED,
        },
        status: ECoverageCheckStatus.ERROR,
        summary: {
          functions: {
            expected: 50,
            current: 20,
            status: ECoverageCheckStatus.ERROR,
          },
        },
      },
    })
  ),
};
