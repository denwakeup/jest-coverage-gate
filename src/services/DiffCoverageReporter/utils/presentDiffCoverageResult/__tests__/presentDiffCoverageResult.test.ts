import jestSnapshotSerializerAnsi from 'jest-snapshot-serializer-ansi';

import { diffCoverageCheckResultMock, rootDirMock } from '../../../../../mocks';
import { presentDiffCoverageResult } from '../index';

expect.addSnapshotSerializer(jestSnapshotSerializerAnsi);

describe('Diff Coverage Result presenter', () => {
  it('should work correctly', () => {
    expect(
      presentDiffCoverageResult({
        result: diffCoverageCheckResultMock,
        rootDir: rootDirMock,
        threshold: {
          added: {
            functions: 50,
            branches: 50,
            lines: 50,
            statements: 50,
          },
          modified: {
            functions: 20,
            branches: 20,
            lines: 20,
            statements: 20,
          },
        },
      })
    ).toMatchSnapshot();
  });
});
