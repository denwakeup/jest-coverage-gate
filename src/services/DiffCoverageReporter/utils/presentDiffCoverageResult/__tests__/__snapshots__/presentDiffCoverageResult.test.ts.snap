// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diff Coverage Result presenter should work correctly 1`] = `
"
Diff coverage quality gate failed:

┌─────────────────────────────────────────────┬─────────┬──────────┬─────────┬─────────┬─────────────┐
│ File                                        │ % Stmts │ % Branch │ % Funcs │ % Lines │ File status │
├─────────────────────────────────────────────┼─────────┼──────────┼─────────┼─────────┼─────────────┤
│ apps/simple/src/utils/parseData.ts          │ 0       │ 0        │ 80      │ 0       │ MODIFIED    │
├─────────────────────────────────────────────┼─────────┼──────────┼─────────┼─────────┼─────────────┤
│ apps/simple/src/components/Header/index.tsx │ 0       │ 0        │ 20      │ 0       │ ADDED       │
└─────────────────────────────────────────────┴─────────┴──────────┴─────────┴─────────┴─────────────┘

Diff coverage threshold for modified files:
functions - 20%
branches - 20%
lines - 20%
statements - 20%

Diff coverage threshold for added files:
functions - 50%
branches - 50%
lines - 50%
statements - 50%
"
`;
